### ISSUE TITLE
Introduce a chapter on quantum information in Error correction unit 

### ISSUE THREAD

**Issue by AnDa-creator** [qiskit-community/qiskit-textbook#652]
**What should the reader gain from this new content?**

I want to suggest a chapter on quantum information. After reading the chapter the student will understand about other representations of state offered by Qiskit like density matrices. Also, learn a little bit about quantum channel' representation like Kraus, Superoperators, Stinespring etc (One or two would be enough). Also a little bit can be introduced about shanon entropy. Basically after reading this chapter, the student will be confidently able to learn about and apply Qiskit terra Quantum information module.

**Where do you expect this new content to fit in with the current content?**

The current content is highly interrelated to this. In error mitigation we use noise simulators which correspond to a randomized kraus operator matrix. It would be a great insight for students to understand about quantum channels and adiabatic computing.

**What material should this new content be based on?**

First and foremost, the documentation is well written for advanced learners, but still can be used. 
(https://qiskit.org/documentation/apidoc/quantum_info.html#quantum-information-qiskit-quantum-info)
As with direct practical implications, this chapter would be good for appendix or error correction I think. 
Unit 3 on Quantum Information in the book by Nielson-chuang titled "Quantum computation and Quantum Information" would provide enough theoretical information to write the chapters. 
Also, video: "Circuit Sessions with Christopher Wood" (https://www.youtube.com/watch?v=VvP41TwY34o) would be helpful to create practical implementations.

**Are you proposing to write this new content?** (y/n)
y (collaboration would be great)

**Comment by AnDa-creator** [comment]
Hi @frankharkins, can you assign me on this ?

**Comment by frankharkins** [comment]
This is a good idea and is already in the works, see: #514 and #8.

**Comment by AnDa-creator** [comment]
@frankharkins, went through the above issues. I liked the new chapter on density matrices and mixed states. As mentioned there, Schimdt decomposition is also as important as partial tracing to get a quantitative idea for pure and mixed states. Also, purification of states is also important. Regarding this the concept of SVD matrices would be important too. So, I would like to add on Schimdt decomposition and purification of states. 
There are many theoretical references but it will be needed to think in terms of Qiskit implementation. So I think I would be able to start working on it. Later as the chapter grows, I think concepts of quantum channels and entropies can be built on these basics.

**Comment by frankharkins** [comment]
Do you imagine this as the next chapter after the one on density matrices?

**Comment by AnDa-creator** [comment]
Yes there can be a small chapter on Schimdt decomposition and purification in the QI unit, it would be needed.

**Comment by frankharkins** [comment]
Okay great, please propose the contents of the chapter ([like in this comment](https://github.com/qiskit-community/qiskit-textbook/issues/653#issuecomment-672767435)).

**Comment by AnDa-creator** [comment]
Yes. I think the chapter can comprise of the following in order:

- Basic composite quantum systems and reasons for constituting a composite quantum system.
- Importance of Schmidt decomposition to find properties of pure states for composite systems.(Schmidt numbers)
- Eigen decomposition and singular value decomposition and relation between them.
- Schmidt decomposition intutive idea and proof.
- Purification of a system by making it composite with another system.(Needs concepts mentioned above)
- Final touch on quantifying the amount of entanglement present in a system using von-neumann entrpy.
- Will contain small exercises and try to keep it less rigorous.
Also, discussing with diemilio about ideas on presenting the chapter.

**Comment by diemilio** [comment]
QI

Hi @AnDa-creator and @frankharkins. I have a suggestion and was wondering what you all think. 

For some time now, I have been thinking about some of the things that I could potentially contribute to the QI chapter once I am done with Density Matrix section. I have put together an outline in my mind of how things should look like, but had been waiting for the Qiskit Textbook team to decide how the chapter was going to be structured before proposing where I could help.

To me, I think the following subjects would be of great interest, and can help readers familiarize themselves both with concepts related to quantum information, and with how Qiskit can be used as a tool to learn about them. Here are some of the sections I think would be of value to include within the QI chapter:

- Short intro to QI
    - This would be just a quick description of what QI is, and why is important to learn about it. The format could be similar to that of the intros to [chapters 1](https://qiskit.org/textbook/ch-states/introduction.html) and [2](https://qiskit.org/textbook/ch-gates/introduction.html).
 
- The Density Matrix and Mixed States
    - This is the section I am currently working on. Here is what I have included so far:
        - Description of the density matrix
        - Representation of pure states via the density matrix
        - Formal definition of mixed states and their density matrix representation
        - Properties of the density matrix
        - The reduced density matrix
        - Visualization of mixed states in the Bloch sphere
    - I think the subjects I have included therein should be sufficient, but if I am open to suggestions if anyone believes there is anything missing.

- Quantum Operations and Noise Channels
    I believe Qiskit’s quantum_info module has several useful tools related to this subject, so I think this might be of interest to many readers. Here are some of the subjects that could be included:
        - Introduction to Quantum operations and noise
            - This can include a subsection in purification, since here is where that tool is of the most use.
        - Basic Noise channels. Can include the basic descriptions of dephasing and depolarizing channels, and combinations of them. Examples of these can possibly be implemented with Qiskit
        - Descriptions of the different channels supported by Qiskit (SuperOperators, Kraus operators, etc.), with examples.

- Benchmarking of states and gates
   Currently, Chapter 5 of the Qiskit textbook says that state and process tomography will soon be added. I think those subjects will fit well within the QI chapter. The section could include:
        - Description of quantum state tomography
        - Quantum state fidelity
        - Quantum gate fidelity (requires the definition of system-environment models, so this section should probably go after the Quantum Operations and Noise Channels section)
            - Minimum gate fidelity
            - Entanglement gate fidelity
            - Average gate fidelity
        - Process tomography
        - Randomized benchmarking. This section already exists in the textbook, but some of the concepts are a bit hard to follow without the proper background. I think adding more context to it might be helpful

- Quantum Entanglement
    - Intro to quantum entanglement. Describes that systems can not just be maximally-entangled or separable, but that there are degrees of entanglement
    - Schmidt decomposition to explain how bipartite/multi-partite states can be decomposed as sums of orthonormal states of each subsystem
    - Quantifying entanglement. Describe measures of entanglement. These include 1) the use of the Schmidt number (for which Schmidt decomposition is needed), and 2) Von Neumann Entropy. In here, it might be necessary to briefly discuss Shannon entropy before introducing Von Neumann entropy
    - Discuss the concept of entanglement as a resource. Explain how entanglement is an actual physical resource (just like energy), and how this can be proven by showing the "fungibility" of entanglement.

**Comment by AnDa-creator** [comment]
Hi @frankharkins and @diemilio, I think the outline planned here is great and most probably subject to verification of the actual textbook team.
But I think it would be best if I start on the topics I proposed and maybe reorganize later based on the plan.

**Comment by frankharkins** [comment]
@quantumjim what do you think?

**Comment by tgag17** [comment]
Should different types of measurements like POVMs be included in the proposed section

**Comment by diemilio** [comment]
Hi @tgag17, I think that is a great question!

I think at this point we are waiting to see what the Qiskit Textbook team thinks makes sense to include in the Quantum Information chapter (if it is decided that there will be a chapter at all).

The outline I suggested above was basically what I (as a student) would like learn about through the textbook by the use of Qiskit. I suggested those topics because I think Qiskit's `quantum_info` module has really great tools to enhance the learning process of those particular subjects, but I am sure there is a lot more that could be added and explored! (like, as you suggested, POVMs).

**Comment by epelaaez** [comment]
@diemilio @AnDa-creator @frankharkins was it decided if the chapter on QI will be worked on? I couldn't found any further issue or PR concerning this. If the QI chapter is still planned, I would like to know if there has been any progress so I can collaborate :)

**Comment by diemilio** [comment]
Hi @epelaaez. How are you? Cool name!! As far as I know, there currently isn’t anyone working on the rest of the chapter. The Density Matrix section that I worked on was merged a little while ago, but it is under the hardware chapter right now. I think that having the density matrix is a good start, and it would be definitely worth adding more sections to finally have a chapter on QI. It is obviously up to @frankharkins and the rest of the qiskit-textbook team if this makes sense, and what the outline should look like, but I would think what I proposed above can at least serve as a skeleton for how the chapter could be structured. 

Do you have interest in working on any of these subjects? I would be willing to collaborate if time allows

**Comment by epelaaez** [comment]
HI @diemilio. Yes, I'm interested in working on the **Benchmarking of states and gates** described on the structure you provided. I think the outline you provided sounds good. I just did a few changes I think would be good, so my proposal looks like:

- Background (could be based on section *II. Notation and Background* of the paper [*Gate fidelity fluctuations and quantum process invariants*](https://arxiv.org/abs/0910.1315)).
- Description of quantum state tomography
     - Demo in Qiskit following the paper [*Maximum Likelihood, Minimum Effort*](https://arxiv.org/abs/1106.5458).
     - Talk about variational approach from [*Variational Quantum Circuits for Quantum State Tomography*](https://arxiv.org/abs/1912.07286).
- Quantum state fidelity
    - Relation to the SWAP test 
- Quantum gate fidelity (requires the definition of system-environment models, so this section should probably go after the Quantum Operations and Noise Channels section)
    - Minimum gate fidelity
    - Entanglement gate fidelity
    - Average gate fidelity
     - Process tomography
- Randomized benchmarking. This section already exists in the textbook, but some of the concepts are a bit hard to follow without the proper background. I think adding more context to it might be helpful

Let me know if these changes to the outline sound good or what could be improved. And I'd be happy to collaborate on this if possible.

**Comment by frankharkins** [comment]
@epelaaez thanks for offering to contribute, you can start working on the background section. As soon as you have a first draft, please make a PR so we can give early feedback.

**Comment by frankharkins** [comment]
Not sure if this is still being worked on, but quite a lot has changed with the new textbook beta release. Unfortunately we're not accepting new content at the moment, but I hope you continue to work on this and share via other channels.

I'm going to close this issue, but let me know if I can support in any way.
